iwallet/block.go

```diff
@@ -22,23 +22,24 @@ import (
 	"github.com/spf13/cobra"
 )
 
-var method string
+var byNum bool
 var complete bool
 
-// blockCmd represents the block command
+// blockCmd represents the block command.
 var blockCmd = &cobra.Command{
 	Use:   "block",
-	Short: "print block info",
-	Long:  `print block info, find by block number or hash`,
-	Args:  cobra.ExactArgs(1),
-	RunE: func(cmd *cobra.Command, args []string) (err error) {
+	Short: "Print block info",
+	Long:  `Print block info by block number or hash`,
+	Args: func(cmd *cobra.Command, args []string) error {
 		if len(args) < 1 {
-			fmt.Println(`Error: block num or hash not given`)
-			return
+			return fmt.Errorf("Please enter the block number or hash")
 		}
+		return nil
+	},
+	RunE: func(cmd *cobra.Command, args []string) (err error) {
 		var blockInfo *rpcpb.BlockResponse
 		var num int64
-		if method == "num" {
+		if byNum {
 			num, err = strconv.ParseInt(args[0], 10, 64)
```

Comment:
I suggest to keep the `method` and use `switch` to judge.

Reply:
agree

Reply:
OK, just think using strings would lead to typo easily. Would we need future methods?